src/hlc_main.c

```diff
@@ -21,6 +21,10 @@
  */
 #include <hlc.h>
 
+#ifdef HL_MOBILE
+#   include <SDL_main.h>
+#endif
+
```

Comment:
This one still needs to go away, what is it for actually?

Reply:
By including SDL_main.h we wrap the main() entry point with SDL code handling glue code for iOS app startup (app delegate).

We could skip including this header,  and use SDL_SetMainReady() function before SDL_Init() call into libs/sdl/sdl.c to circumvent SDL_Init() failing (it's just a stupid boolean check), but sadly it is not enough.

We would need to basically to copy/paste the SDL iOS AppDelegate code from SDL2 source code or rewrite a similar one to get a proper iOS app context. Another solution could be to copy/paste only SDL_uikitappdelegate.h from the SDL source tree, subclass it, and in hlc_main redefine main to SDL_main if we are on HL_MOBILE.

See https://github.com/spurious/SDL-mirror/blob/master/src/video/uikit/SDL_uikitappdelegate.m for more details on what the app delegate does.

Pretty similar situation on Android with SDL_main "magic" wrapping the SDLActivity Java class.

Feel free to share any better idea.

Reply:
Ok, I think we can workaround this by simply checking `#if defined(HL_MOBILE) && defined(sdl__Sdl__val)` , this way this does not imply that HL_MOBILE always uses SDL

Reply:
Okay, just fixed that in latest commit.